From 8d71d4fc2595faf8e7ef7c1e7e77065bf5f59a92 Mon Sep 17 00:00:00 2001
From: Ramasamy Kaliappan <quic_rkaliapp@quicinc.com>
Date: Thu, 2 Nov 2023 16:17:13 +0530
Subject: [PATCH] wpa_supplicant: Add open mode support in MLD STA

Allow encryption none for mld STA.

NOTE: Need to double check spec. This patch is not available in upstream

Signed-off-by: Ramasamy Kaliappan <quic_rkaliapp@quicinc.com>
---
 wpa_supplicant/events.c | 3 +++
 1 file changed, 3 insertions(+)

--- a/wpa_supplicant/events.c
+++ b/wpa_supplicant/events.c
@@ -3523,6 +3523,9 @@ static int wpa_sm_set_ml_info(struct wpa
 	const u8 *bss_rsn = NULL, *bss_rsnx = NULL;
 	int i;
 
+	if (wpa_s->key_mgmt == WPA_KEY_MGMT_NONE)
+		return 0;
+
 	os_memset(&drv_mlo, 0, sizeof(drv_mlo));
 	if (wpas_drv_get_sta_mlo_info(wpa_s, &drv_mlo)) {
 		wpa_dbg(wpa_s, MSG_INFO, "Failed to get MLO link info");
